src/modules/cpu/kernel/image_sum.hpp

```diff
@@ -0,0 +1,870 @@
+                }
+            }
+            sum = sumR + sumG + sumB;
+            imageSumArr[batchCount * 4] = (Rpp32f)sumR;
+            imageSumArr[(batchCount * 4) + 1] = (Rpp32f)sumG;
+            imageSumArr[(batchCount * 4) + 2] = (Rpp32f)sumB;
+            imageSumArr[(batchCount * 4) + 3] = (Rpp32f)sum;
+        }
+
+        // Image Sum without fused output-layout toggle (NHWC)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC))
+        {
+            Rpp64f sum, sumR = 0.0, sumG = 0.0, sumB = 0.0;
+            Rpp64f sumAvxR[4] = {0.0};
+            Rpp64f sumAvxG[4] = {0.0};
+            Rpp64f sumAvxB[4] = {0.0};
+
+            for(int c = 0; c < layoutParams.channelParam; c++)
+            {
+                Rpp8u *srcPtrRow;
+                srcPtrRow = srcPtrChannel;
+#if __AVX2__
+                __m256d psumR = _mm256_setzero_pd();
+                __m256d psumG = _mm256_setzero_pd();
+                __m256d psumB = _mm256_setzero_pd();
+#endif
+                for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+                {
+                    Rpp8u *srcPtrTemp;
+                    srcPtrTemp = srcPtrRow;
+
+                    int vectorLoopCount = 0;
+                    for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                    {
+#if __AVX2__
+                        __m256d p[6];
+                        rpp_simd_load(rpp_load24_u8pkd3_to_f64pln3_avx, srcPtrTemp, p);
+                        compute_sum_24_host(p, &psumR, &psumG, &psumB);
+#endif
+                        srcPtrTemp += vectorIncrement;
+                    }
+                    for (; vectorLoopCount < bufferLength; vectorLoopCount += 3)
+                    {
+                        sumR += (Rpp64f)(srcPtrTemp[0]);
+                        sumG += (Rpp64f)(srcPtrTemp[1]);
+                        sumB += (Rpp64f)(srcPtrTemp[2]);
+                        srcPtrTemp += 3;
+                    }
+                    srcPtrRow += srcDescPtr->strides.hStride;
+                }
+                srcPtrChannel += srcDescPtr->strides.cStride;
+#if __AVX2__
+                rpp_simd_store(rpp_store4_f64_to_f64_avx, sumAvxR, &psumR);
+                rpp_simd_store(rpp_store4_f64_to_f64_avx, sumAvxG, &psumG);
+                rpp_simd_store(rpp_store4_f64_to_f64_avx, sumAvxB, &psumB);
+#endif
+                for(int i = 0; i < 2; i++)
+                {
+                    sumR += (sumAvxR[i] + sumAvxR[i + 2]);
+                    sumG += (sumAvxG[i] + sumAvxG[i + 2]);
+                    sumB += (sumAvxB[i] + sumAvxB[i + 2]);
+                }
+            }
+            sum = sumR + sumG + sumB;
+            imageSumArr[batchCount * 4] = (Rpp32f)sumR;
+            imageSumArr[(batchCount * 4) + 1] = (Rpp32f)sumG;
+            imageSumArr[(batchCount * 4) + 2] = (Rpp32f)sumB;
+            imageSumArr[(batchCount * 4) + 3] = (Rpp32f)sum;
```

Comment:
Pls precompute these batchCount * 4 in advance

Reply:
Done